### PR TITLE
Update version check for deprecated getCallState and getNetworkType

### PR DESCRIPTION
**Pull request by amytsengtw**
getCallState was deprecated, use getCallStateForSubscription to get the call state.

https://developer.android.com/reference/android/telephony/TelephonyManager#getCallState%28%29

- - -
This change is [Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/180)

### CODE REVIEW

File: src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public String getSubscriberId() {
                     "Gets the call state for the default subscription. Call state values are"
                             + "0: IDLE, 1: RINGING, 2: OFFHOOK")
     public int getTelephonyCallState() {
-        return mTelephonyManager.getCallState();
+        return mTelephonyManager.getCallStateForSubscription();
```
Comment (ko1in1u) [inline src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java:50]: `getCallStateForSubscription` is added in SDK 31, this change might break the usage on older SDK. Please add a check here.

```
if (Build.VERSION.SDK_INT < 31) {  # or use Build.VERSION_CODES.S
    return mTelephonyManager.getCallState();
} else {
    return mTelephonyManager.getCallStateForSubscription();
}
```

Comment (amytsengtw) [reply src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java:50]: add

File: src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java [hunk 2]
```diff
@@ -47,15 +47,19 @@ public String getSubscriberId() {
...
```
Comment (ko1in1u) [inline src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java:50]: this one also need to add a check for devices running SDK lower than 31

File: src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java [hunk 3]
```diff
@@ -55,7 +55,11 @@ public int getTelephonyCallState() {
                     "Returns a constant indicating the radio technology (network type) currently"
                             + "in use on the device for data transmission.")
     public int getDataNetworkType() {
-      return mTelephonyManager.getDataNetworkType();
+        if (Build.VERSION.SDK_INT < 31) {
+            return mTelephonyManager.getDataNetworkType();
+        } else {
+            return mTelephonyManager.getCallStateForSubscription();
```
Comment (ko1in1u) [inline src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java:61]: this is not related to data network type

File: src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java (hunk 2 above)
Comment (amytsengtw) [reply src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java]: done

File: src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java (hunk 3 above)
Comment (amytsengtw) [reply src/main/java/com/google/android/mobly/snippet/bundled/TelephonySnippet.java]: [getNetworkType](https://developer.android.com/reference/android/telephony/TelephonyManager#getNetworkType())() was deprecated in API level 30, add check for version < 30

**Review by ko1in1u — APPROVED** [review]
